modules/spring-config/src/main/java/org/mule/runtime/config/internal/LazyMuleArtifactContext.java

```diff
@@ -295,14 +337,14 @@ public void initializeComponents(Predicate<org.mule.runtime.config.internal.mode
             .initializeComponents(componentModelPredicate1, applyStartPhase));
   }
 
-  private List<String> createComponents(Optional<Predicate> predicateOptional, Optional<Location> locationOptional,
-                                        Optional<ComponentModelInitializerAdapter> parentComponentModelInitializerAdapter) {
+  private Map<String, Object> createComponents(Optional<Predicate> predicateOptional, Optional<Location> locationOptional,
```

Comment:
![Code Smell](https://sonarsource.github.io/sonar-github/code_smell.png 'Code Smell') Code Smell: Refactor this method to reduce its Cognitive Complexity from 19 to the 15 allowed. ([squid:S3776](https://sonarqube.kbuild.msap.io/coding_rules#rule_key=squid%3AS3776))

[See it in SonarQube](https://sonarqube.kbuild.msap.io/project/issues?id=mule-runtime-mule&pullRequest=7761&issues=AWnZSM1wotIXnmsU4589&open=AWnZSM1wotIXnmsU4589)

[ISSUE_KEY]: AWnZSM1wotIXnmsU4589